### PR TITLE
cxxtest: optional stream test

### PR DESCRIPTION
**Pull request by cuiziwei1**
## Summary

skip the locale related test if libc doesn't enable CONFIG_CXX_LOCALIZATION.

## Impact

cxxtest

## Testing

ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



I need the content of the PR to assess if it meets the requirements. Please provide the following:

* **The actual changes made in the PR:**  A link to the PR or a description of the code modifications.
* **The PR description:** This usually contains information about the purpose and details of the changes. 

With this information, I can then evaluate the PR against the NuttX requirements and provide you with a concise answer.

**Review by cederom — CHANGES_REQUESTED** [review]
This is CPP test, you cannot remove C++ specific calls here, rejecting.

PR without description will be rejected, see https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md.

If you think otherwise please reopen and explain the concept in detail.

**Comment by xiaoxiang781216** [comment]
> This is CPP test, you cannot remove C++ specific calls here, rejecting.
> 

why? locale is an optional feature in nuttx libc, so c++ test code should skip locale related test if libc close this feature.

> PR without description will be rejected, see https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md.
> 
> If you think otherwise please reopen and explain the concept in detail.

**Comment by cederom** [comment]
Thanks @xiaoxiang781216 :-) 

It is reaaaallly important to write good description of the PR, there was none. Contributing Guidelines were updated to provide examples, there is a reference at top of PR report, we even created PR AI BOT to suggest how the description can be improved to help in quick/good assessment. We have a lot of PR to analyze so we need the description why change is necessary, what it fixes/adds, what is the testing result before and after, so PR dont break more than it fixes. We cannot blindly accept undocumented PRs because things start falling apart :-)

C++ NuttX maintainers would be the best person to ask for a review here, I don't work with C++ sorry :-)

What I don't like here is replacing `std::cout` (C++ way) with `printf` (C way) and there is no explanation why this is necessary. I know C++ folks don't like this kind of stuff. It was here before for a reason right? :-)

File: testing/cxxtest/cxxtest_main.cxx [hunk 1]
```diff
@@ -128,26 +131,29 @@ static void test_ofstream()
...
+#endif
 
 //***************************************************************************
 // Name: test_stl
 //***************************************************************************/
 
 static void test_stl()
 {
-  std::cout << "Test std::vector =============================" << std::endl;
+  printf("Test std::vector =============================\n");
```
Comment (gustavonihei) [inline testing/cxxtest/cxxtest_main.cxx:156]:
```suggestion
  std::printf("Test std::vector =============================\n");
```
For consistency to the other uses of `std::printf` in this same source file.

**Comment by xiaoxiang781216** [comment]
> Thanks @xiaoxiang781216 :-)
> 
> It is reaaaallly important to write good description of the PR, there was none. Contributing Guidelines were updated to provide examples, there is a reference at top of PR report, we even created PR AI BOT to suggest how the description can be improved to help in quick/good assessment. We have a lot of PR to analyze so we need the description why change is necessary, what it fixes/adds, what is the testing result before and after, so PR dont break more than it fixes. We cannot blindly accept undocumented PRs because things start falling apart :-)
> 

Yes, it always good to request the contributor to add the detailed description.

> C++ NuttX maintainers would be the best person to ask for a review here, I don't work with C++ sorry, if they give a GO you have my GO too :-)

Most C++ change in NuttX contributes by my team, so I am a good candidate to review the related change.

> 
> What I don't like here is replacing `std::cout` (C++ way) with `printf` (C way) and there is no explanation why this is necessary. 

Because cin/cout depends on locale to pass the build, that's why to replace the code which call cin/cout unconditionally, the conditional part still keep as before:
https://github.com/apache/nuttx-apps/pull/2722/files#diff-9d02d89b087be466dd6bbdf41f260690b28365e09a162260f69e20ce7a4e2896R110
https://github.com/apache/nuttx-apps/pull/2722/files#diff-9d02d89b087be466dd6bbdf41f260690b28365e09a162260f69e20ce7a4e2896R141

> I know C++ folks don't like this kind of stuff. It was here before for a reason right? :-)

cout/cin is very code size consumption:
https://github.com/apache/nuttx-apps/blob/master/testing/cxxsize/README.md
it's good to avoid iostream in embedded space.

**Comment by xiaoxiang781216** [comment]
Yes, @cuiziwei1 please update the pr, it's always important to describe the change clearly.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**